Flux monitor test should fail on ci

RyanRHall opened the pull request:
All CI tests were passing because the the jest tests were wrapped in `set +e`

This PR makes the exit code of `compost test:ts` that of the jest test

It also makes the test use `MINIMUM_CONTRACT_PAYMENT` env var, squashing an easy bug

Firstly identified, is this [bug](https://www.pivotaltracker.com/story/show/172085379)

j16r commented on tools/docker/compose:
```diff
@@ -119,12 +119,14 @@ case "$1" in
     $deps up --exit-code-from wait-db wait-db
     $deps up --exit-code-from wait-db-2 wait-db-2
     set +e
```
Wouldn't it be better not to use `set +e` ?

spooktheducks replied:
I think `set +e` is the default -- we want `set -e`, don't we?

j16r replied:
Yeah, should be okay to error out if any of those parts fail.

RyanRHall replied:
default behavior is `set +e` .. we use a `set -e` at the top of the script because we want to error ASAP if anything goes wrong in the setup / tear down or really any other part of the script. This is the only place where we expect a 1 exit code might occur but don't want to exit immediately, so we wrap it in a
```
set +e
...
set -e
```

coventry replied:
Small nit, but it's more idiomatic and less error-prone to suffix the command with `|| /bin/true`.

spooktheducks replied:
Ah, this makes sense -- otherwise we'd lose the test logs

RyanRHall replied:
Correct @spooktheducks

RyanRHall commented:
@coventry Personally I'd rather just create a separate story for addressing the feedback from shellcheck since it does not address the changes in this PR and feels a bit out of scope.

spooktheducks commented:
> @coventry Personally I'd rather just create a separate story for addressing the feedback from shellcheck since it does not address the changes in this PR and feels a bit out of scope.

I agree -- I'll throw a story in tracker.

RyanRHall commented:
@j16r we need to capture the exit code of the jest test because the script needs to exit with the same code. Appending an `|| true` would make it so that the test never fails.
